src/core/vectortile/qgsmapboxglstyleconverter.cpp

```diff
@@ -453,7 +456,7 @@ bool QgsMapBoxGlStyleConverter::parseFillLayer( const QVariantMap &jsonLayer, Qg
     fillSymbol->setStrokeStyle( Qt::NoPen );
   }
 
-  if ( fillColor.isValid() )
+  if ( fillColor.isValid() || colorIsDataDefined )
   {
     fillSymbol->setFillColor( fillColor );
```

Comment:
If fillColor isn't valid, then we'll be setting the fill symbol to an invalid QColor here... I don't think we want to do that!

Reply:
What would you do? Move the check and do an else if?

Reply:
Can you explain what it's trying to do here? I think ideally it'd be set to eg the first colour in the list

Reply:
I'm just trying to avoid to set "no brush" if it's data defined. So not setting it might be sufficient.